python/pylibcudf/pylibcudf/libcudf/datetime.pxd

```diff
@@ -1,12 +1,26 @@
 # Copyright (c) 2020-2024, NVIDIA CORPORATION.
 
+from libc.stdint cimport int32_t
 from libcpp.memory cimport unique_ptr
 from pylibcudf.libcudf.column.column cimport column
 from pylibcudf.libcudf.column.column_view cimport column_view
 from pylibcudf.libcudf.scalar.scalar cimport scalar
 
 
 cdef extern from "cudf/datetime.hpp" namespace "cudf::datetime" nogil:
+    cpdef enum class datetime_component(int32_t):
```

Comment:
As @bdice suggests, make the underlying type explicit in the C++ and then match it here.